crates/ruff_python_resolver/src/implicit_imports.rs

```diff
@@ -128,14 +128,14 @@ pub(crate) fn find(dir_path: &Path, exclusions: &[&Path]) -> HashMap<String, Imp
 
 /// Filter a map of implicit imports to only include those that were actually imported.
 pub(crate) fn filter(
-    implicit_imports: &HashMap<String, ImplicitImport>,
+    implicit_imports: &BTreeMap<String, ImplicitImport>,
```

Comment:
Nit: Unrelated to this PR. We should probably wrap these types that we pass around frequently. It would reduce the scope of changes to internal data structures and could even allow us to implement some of these functions as methods.